.github/workflows/codeql.yml

```diff
@@ -0,0 +1,60 @@
+name: "CodeQL"
+
+on:
+    push:
+        branches: ["main"]
+    pull_request:
+        branches: ["main"]
+    schedule:
+        - cron: "37 18 * * 6"
+
+jobs:
+    analyze:
+        name: Analyze (${{ matrix.language }})
+        runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
+        permissions:
+            # required for all workflows
+            security-events: write
+
+            # required to fetch internal or private CodeQL packs
+            packages: read
+
+            # only required for workflows in private repositories
+            actions: read
+            contents: read
+
+        strategy:
+            fail-fast: false
+            matrix:
+                include:
+                    - language: csharp
+                      build-mode: none
+                    - language: go
+                      build-mode: autobuild
+                    - language: java-kotlin
+                      build-mode: none # This mode only analyzes Java. Set this to 'autobuild' or 'manual' to analyze Kotlin too.
+                    - language: javascript-typescript
+                      build-mode: none
+                    - language: python
+                      build-mode: none
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v4
+
+            # Initializes the CodeQL tools for scanning.
+            - name: Initialize CodeQL
+              uses: github/codeql-action/init@v3
+              with:
+                  languages: ${{ matrix.language }}
+                  build-mode: ${{ matrix.build-mode }}
+                  config-file: .github/codeql/codeql-config.yml
+
+            - if: matrix.build-mode == 'manual'
+              shell: bash
+              run: |
+                  echo 'TODO: build commands for all java and kotlin codes.'
+
```

Comment:
fix
use `./gradlew jar` for example
I think if condition should check for language, not for mode

Reply:
future maybe?

Reply:
Java client isn't built and skipped as I understand